lib/Common/Codex/Utf8Codex.cpp

```diff
@@ -395,10 +394,11 @@ namespace utf8
             cch -= 4;
         }
 LSlowPath:
+        // WARNING: Assume cch correct, suppress end-of-buffer checking
+        // we might have failed to measure cch, +1 for odd length. sizeof(char16) == 2
+        LPCUTF8 end = ptr + cch + (cch % 2 ? 2 : 1);
         while (cch-- > 0)
         {
-            LPCUTF8 end = ptr + cch + 1; // WARNING: Assume cch correct, suppress end-of-buffer checking
```

Comment:
@jianchun this algorithm works differently. If you look at closer, it is almost the same with original one and this one saves reading 1 extra byte